portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -19,53 +19,73 @@
 import com.google.appengine.api.datastore.Entity;
 import com.google.appengine.api.datastore.PreparedQuery;
 import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
 import com.google.gson.Gson;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import java.util.ArrayList;
+import java.util.List;
+import java.util.Collections;
 
 
 /**
-  * Servlet that returns some example content. 
+  * Servlet that returns comments in database. 
   * Note: We first added the Gson library dependency to pom.xml.
   */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-
+  int maxNumComments = 1;
+  ArrayList<String> msglist = new ArrayList<String>();
   String comment = "Comment";
+  String contentProperty = "content";
+  String timestampProperty = "timestamp";
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String inputNumComment = "numComment";
+    maxNumComments = Integer.parseInt(request.getParameter(inputNumComment));
```

Comment:
change this to something like. 
int maxNumComments = defaultNumComments.
If(request.getParameter(inputNumComment) != null ){
maxComments = ...
}
That way you aren't editing the global variable and can make it a constant.